doc/sphinx_source/tutorials/module.rst

```diff
@@ -0,0 +1,234 @@
+
+Similar to adding a partyline command, you also have to create a function table for a new Tcl command::
+
+  static tcl_cmds mytcl[] = {
+    {"twcmd",           tcl_twcmd},
+    {NULL,                   NULL}   /* Required to mark end of table */
+  };
+
+Adding a Tcl Bind
+-----------------
+
+A Tcl bind is a command that is activated when a certain condition is met. With Eggdrop, these are usually linked to receiving messages or other IRC events. To create a bind, you must first register the bind type with Eggdrop when the module is loaded (you added the woobie_start() and woobie_close functions earlier, you still need all that earlier code in here as well)::
+
+  static p_tcl_bind_list H_woob;
+
+  ...
+
+  char *woobie_start(Function *global_funcs)
+  {
+    ...
+    H_ccht = add_bind_table("woobie", HT_STACKABLE, woobie_2char);  
+  }
+
+And then remove the binds when the module is unloaded::
+
+  static char *twitch_close()
+  {
+    ...
+    del_bind_table(H_ccht);
+  }
+
+Here, "woobie" is the name of the bind (similar to the PUB, MSG, JOIN types of binds you already see in tcl-commands.doc). HT_STACKABLE means you can have multiple binds of this type. "woobie_2char" defines how many arguments the bind will take, and we'll talk about that next.
+
+Defining bind arguments
+^^^^^^^^^^^^^^^^^^^^^^^
+
+The following code example defines a bind that will take two arguments::
+
+  static int woobie_2char STDVAR
+  {
+    Function F = (Function) cd;
+
+    BADARGS(3, 3, " nick chan");
+
+    CHECKVALIDITY(woobie_2char);
+    F(argv[1], argv[2]);
+    return TCL_OK;
+  }
+
+And this example defines a bind that will take three arguments::
+
+  static int woobie_3char STDVAR
+  {
+    Function F = (Function) cd;
+
+    BADARGS(4, 4, " foo bar moo");
+
+    CHECKVALIDITY(woobie_3char);
+    F(argv[1], argv[2], argv[3]);
+    return TCL_OK;
+  }
+
+Like before, BADARGS still checks that the number of arguments passed is correct, and outputs help text if it is not. The rest is boilerplate code to pass the arguments when the bind is called.
+
+Calling the Bind
+^^^^^^^^^^^^^^^^
+
+To call the bind, Eggdrop coding style it to name that function "check_tcl_bindname". So here, whenever we reach a point in code that should trigger the bind, we'll call check_tcl_woobie() and pass the arguments we defined- in this case, two arguments that woobie_2char was created to handle. Here is some sample code::
+
+  check_tcl_woobie(chan, nick);
+
+
+  static int check_tcl_woobie(char *chan, char *nick) {
+    int x;
+    char mask[1024];
+    struct flag_record fr = { FR_GLOBAL | FR_CHAN, 0, 0, 0, 0, 0 };
+
+    snprintf(mask, sizeof mask, "%s %s!%s@%s.tmi.twitch.tv",
```

Comment:
why twitch hosts are being used isn't explained, what does the example bind mask match against?